express-notes-api/index.js

```diff
@@ -0,0 +1,148 @@
+const express = require('express');
+const fs = require('fs');
+const app = express();
+const data = require('./data.json');
+const notes = data.notes;
+const notesArray = [];
+const nextId = data.nextId;
+const resError = {};
+
+for (const key in notes) {
+  notesArray.push(notes[key]);
+}
+
+app.get('/api/notes', (req, res) => {
+  res.send(notesArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = parseInt(req.params.id);
+  if (id > 0) {
+    let index = -1;
+    for (let i = 0; i < notesArray.length; i++) {
+      if (notesArray[i].id === id) {
+        index = i;
+        break;
+      }
+    }
+    if (index === -1) {
+      resError.error = `cannot find note with id ${id}`;
+      res.status(404).json(resError);
+    } else {
+      res.status(200).send(notesArray[index]);
+    }
```

Comment:
You can look up values directly on an object without a loop at all. `data.notes[id]`